packages/typespec-python/src/emitter.ts

```diff
@@ -249,6 +249,11 @@ export function getType(context: SdkContext, type: EmitterType): any {
             }
             // need to do discriminator outside `emitModel` to avoid infinite recursion
             handleDiscriminator(context, type, newValue);
+
+            // DPG model is inherited from Mapping, so we don't need handle inheritance from "dict"
+            if (newValue.type === "model" && newValue.parents.length > 0 && newValue.parents[0].type === "dict") {
```

Comment:
shouldn't this code be [here](https://github.com/Azure/autorest.python/blob/4015bf85a33707a3464a27509b50dfe96a1dae83/packages/typespec-python/src/emitter.ts#L803) instead, and you just check that it's not a mapping? If the reason you're moving the logic to here is because tcgc is throwing an error or something like that, we should fix tcgc instead

Reply:
updated